### PR TITLE
Fix dark mode color problem in invitation modal and user menu

### LINKED ISSUE #861: Invitation acceptation component background is white under dark mode
## Describe the bug

Invitation acceptation component background is white under dark mode

## To Reproduce

1. Enable dark mode
2. Accept an invite

## Expected behavior

Message should be readable

## Screenshots

image

## Device information

 

- **Device:** PC
- **OS:** Windows 11
- **Browser:** MS Edge
- **Version:** 98

## Additional context

No

### PR DESCRIPTION
**Pull request by Jordanlelay**
## Work performed

- Handle dark mode in invitation modal
- Handle dark mode in invalid invitation modal
- Extract component from duplicated code
- Fix avatar color icon in dark mode

## Resolved issues

Closes #861

### CODE REVIEW

File: typescript/web/src/components/invitation-manager/invitation-container.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+type CompleteProps = PropsWithChildren<InvitationContainerProps>;
+
+export const InvitationContainer = ({
+  header,
+  details,
+  children,
+}: CompleteProps) => {
+  return (
+    <Center h="full">
+      <chakra.div
```
Comment (bbenoist) [inline typescript/web/src/components/invitation-manager/invitation-container.tsx:18]: While you're at it, can we switch to standard Chakra components?

Comment (Jordanlelay) [reply typescript/web/src/components/invitation-manager/invitation-container.tsx:18]: I can but what's wrong with that ?

Comment (bbenoist) [reply typescript/web/src/components/invitation-manager/invitation-container.tsx:18]: It add extra boilerplate code and complexifies what should have just been an imbrication of VStack, Header and HStack without fixed fontSize, fontWeight, backgrounds, some of the paddings, etc.

Comment (bbenoist) [reply typescript/web/src/components/invitation-manager/invitation-container.tsx:18]: It's not critical but while there is a few dev bandwidth while waiting for SW to be finished, I think that we can take a few time to clean this up 🧹

File: typescript/web/src/components/invitation-manager/invitation-container.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  details: string;
+};
+
+type CompleteProps = PropsWithChildren<InvitationContainerProps>;
+
+export const InvitationContainer = ({
+  header,
+  details,
+  children,
+}: CompleteProps) => {
```
Comment (bbenoist) [inline typescript/web/src/components/invitation-manager/invitation-container.tsx:21]: Braces are useless now

File: typescript/web/src/components/invitation-manager/invitation-container.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        borderRadius="md"
+        boxShadow="lg"
+        maxW="2xl"
+        my="3.75rem"
+      >
+        <Heading fontSize="xl" fontWeight="semibold">
+          {header}
+        </Heading>
+        <Flex>{details}</Flex>
+        <Flex w="100%" justify="flex-end">
```
Comment (bbenoist) [inline typescript/web/src/components/invitation-manager/invitation-container.tsx:38]: There must be a cleaner way than using `w="100%"` no?

File: typescript/web/src/components/invitation-manager/invitation-container.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        p={6}
+        spacing={6}
+        align="start"
+        bg={mode("white", "gray.800")}
+        borderRadius="md"
+        boxShadow="lg"
+        maxW="2xl"
+        my="3.75rem"
+      >
+        <Heading fontSize="xl" fontWeight="semibold">
```
Comment (bbenoist) [inline typescript/web/src/components/invitation-manager/invitation-container.tsx:34]: Why don't you use the as property with an appropriate value?

File: typescript/web/src/components/invitation-manager/invitation-container.tsx (hunk 2 above)
Comment (bbenoist) [inline typescript/web/src/components/invitation-manager/invitation-container.tsx:21]:
```suggestion
export type InvitationContainerProps = PropsWithChildren<{
  header: string;
  details: string;
}>;

export const InvitationContainer = ({
  header,
  details,
  children,
}: InvitationContainerProps) => {
```

File: typescript/web/src/components/invitation-manager/invitation-container.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+          {header}
+        </Heading>
+        <Flex>{details}</Flex>
```
Comment (bbenoist) [inline typescript/web/src/components/invitation-manager/invitation-container.tsx:37]:
```suggestion
        <Text>{details}</Text>
```

File: typescript/web/src/components/invitation-manager/invitation-container.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
+import {
+  Center,
+  Flex,
+  Heading,
```
Comment (bbenoist) [inline typescript/web/src/components/invitation-manager/invitation-container.tsx:4]:
```suggestion
  Heading,
  Text,
```

**Review by bbenoist — CHANGES_REQUESTED** [review]

File: typescript/web/src/components/invitation-manager/invitation-container.tsx (hunk 4 above)
Comment (Jordanlelay) [reply typescript/web/src/components/invitation-manager/invitation-container.tsx:34]: I should've explained it to you last time you made that comment.

`as` doesn't do anything on the style, it's just to include the right tag in the DOM. You can put whatever value you want in `as` it won't change the size.

File: typescript/web/src/components/invitation-manager/invitation-container.tsx (hunk 3 above)
Comment (bbenoist) [reply typescript/web/src/components/invitation-manager/invitation-container.tsx]: Apart from the fact that is ignores the padding and some other props, the parent is a flex container so using `width="100%"` is kind of an heresy. You should be able use `alignSelf="stretch"`  or something like that for such cases.

**Review by bbenoist — CHANGES_REQUESTED** [review]
There is a defect when there is an error in the invitation. Please see my denied Chromatic capture for details.

Also, there remains the suggesion about `Flex width="100%"` within a `VStack` thing which I'd prefer to be fixed. It should just be the matter of switching to the correct prop...

File: typescript/web/src/components/invitation-manager/invitation-container.tsx (hunk 3 above)
Comment (Jordanlelay) [reply typescript/web/src/components/invitation-manager/invitation-container.tsx]: Not in all cases. If the text is small enough, the VStack will shrink
I've used it anyway and put a `w="full"` above it

**Review by Jordanlelay — COMMENTED** [review]
@valerian could you give a look as @bbenoist and I both committed in this PR ?

**Review by bbenoist — APPROVED** [review]

**Pull request merged**